Core/include/Acts/Utilities/Any.hpp

```diff
@@ -0,0 +1,427 @@
+      assert(m_handler && "Handler is null");
+      copyConstruct(other);
+    } else {
+      // @TODO: Support assigning between different types
+      assert(m_handler == other.m_handler);
+      copy(other);
+    }
+    return *this;
+  }
+
+  AnyBase(AnyBase&& other) {
+    if (m_handler == nullptr) {  // this object is empty
+      m_handler = other.m_handler;
+      assert(m_handler != nullptr && "Handler is null");
+      moveConstruct(std::move(other));
+    } else {
+      // @TODO: Support assigning between different types
+      assert(m_handler == other.m_handler);
+      move(std::move(other));
+    }
+  }
+
+  AnyBase& operator=(AnyBase&& other) {
+    if (m_handler == nullptr) {  // this object is empty
+      m_handler = other.m_handler;
+      assert(m_handler && "Handler is null");
+      moveConstruct(std::move(other));
+    } else {
+      // @TODO: Support assigning between different types
+      assert(m_handler == other.m_handler);
+      move(std::move(other));
+    }
+    return *this;
+  }
+
+  operator bool() const { return m_handler != nullptr; }
+
+ private:
+  struct Handler {
+    void (*destroy)(void* ptr) = nullptr;
+    void (*moveConstruct)(void* from, void* to) = nullptr;
+    void (*move)(void* from, void* to) = nullptr;
+    void (*copyConstruct)(const void* from, void* to) = nullptr;
+    void (*copy)(const void* from, void* to) = nullptr;
+    std::size_t typeSize{0};
+  };
+
+  template <typename T>
+  static Handler* makeHandler() {
+    static Handler static_handler = []() {
+      Handler h;
+      if constexpr (!std::is_trivially_destructible_v<T> || sizeof(T) > SIZE) {
+        h.destroy = &destroyImpl<T>;
+      }
+      if constexpr (!std::is_trivially_move_constructible_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.moveConstruct = &moveConstructImpl<T>;
+      }
+      if constexpr (!std::is_trivially_move_assignable_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.move = &moveImpl<T>;
+      }
+      if constexpr (!std::is_trivially_copy_constructible_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.copyConstruct = &copyConstructImpl<T>;
+      }
+      if constexpr (!std::is_trivially_copy_assignable_v<T> ||
+                    sizeof(T) > SIZE) {
+        h.copy = &copyImpl<T>;
+      }
+      h.typeSize = sizeof(T);
+      return h;
+    }();
+    return &static_handler;
+  }
+
+  void destroy() {
+    _ACTS_ANY_DEBUG("Destructor this=" << this << " handler: " << m_handler);
+    if (m_handler != nullptr && m_handler->destroy != nullptr) {
+      void* ptr = m_data.data();
+      if (m_handler->typeSize > SIZE) {
+        // stored on heap: interpret buffer as pointer
+        _ACTS_ANY_DEBUG("Pre-destroy data: " << (void*)m_data.data());
+
+        _ACTS_ANY_DEBUG_BUFFER("-> buffer pre-destroy", m_data);
+
+        ptr = *reinterpret_cast<void**>(m_data.data());
+        _ACTS_ANY_DEBUG("Pre-destroy: " << ptr);
+      }
```

Comment:
I think you have too many copies of this logic around. How about just adding a `void* data()` method that returns `(void*)m_data.data()` in the stack-allocated case and `*(void**)m_data.data()` in the heap-allocated case?

Reply:
That only solves it half of the time, since sometimes i need to be able to assign it too. But I can try to centralize the casting anyway.

Reply:
Indeed, for assignment you'd need the by-reference equivalent.

Reply:
Ok replaced.